### ISSUE TITLE
gd-extension: Error retrieving accented characters

### ISSUE THREAD

**Issue by Loxtir** [2shady4u/godot-sqlite#87]
**Environment:**
 - OS: Windows
 - Godot version: 4.0 alpha 14
 - _godot-sqlite_ version: gd-extension, latest bin from github actions

**Issue description:**
 Accented characters are not retrieved correctly from the database. This worked fine on Godot 3.X

**Steps to reproduce:**
 
```
db.query("CREATE TABLE \"Test\" (\"name\" TEXT NOT NULL, PRIMARY KEY(\"name\"))")
db.query("insert into Test (name) values (\"ÁéÍóÚ\")")
db.query("select * from Test")
print(db.query_result[0]["name"])
```
Expected this to print "ÁéÍóÚ", got "ÃÃ©ÃÃ³Ã" instead. Inspecting the database with DBBrowser shows the value is being saved correctly.

**Minimal reproduction project:**
 
[SqliteTest.zip](https://github.com/2shady4u/godot-sqlite/files/9410111/SqliteTest.zip)

**Additional context**
 I hope the bug isn't on my end, it's my first opened issue

**Comment by 2shady4u** [comment]
Hello @Loxtir 

I would expect this to be an issue related to Godot 4.0 itself, but I'm not sure?
I'll do some research and I'll get back to this...

**Comment by 2shady4u** [comment]
https://github.com/godotengine/godot-cpp/issues/784 might be related?

**Comment by Loxtir** [comment]
So if I understand correctly, if I compile the plugin on my own converting to String::utf8 on this line [https://github.com/2shady4u/godot-sqlite/blob/gd-extension/src/gdsqlite.cpp#L291](url) , it should work?

**Comment by 2shady4u** [comment]
You can definitely leave this open until we figure out the cause of this issue 😄 
I'm currently quite busy with other things so this might take a while...
My apologies for that 🙇

**Comment by Loxtir** [comment]
> [godotengine/godot-cpp#784](https://github.com/godotengine/godot-cpp/issues/784) might be related?

So in case it helps someone, undoing [this change](https://github.com/godotengine/godot-cpp/commit/bf8fc4c53d07cf8ac4d8343364899c197076fbc2#diff-20a0dfd8072691f654c12adb6aecabee9d3247fb99488b37539dab1d967c97b8L77-L79) and compiling godot-cpp and the plugin myself fixes the issue.

**Comment by lightyears1998** [comment]
I encountered a similar issue. By changing [this line in gdsqlite.cpp](https://github.com/2shady4u/godot-sqlite/blob/70646d844e24015c458c90874986b7c46de1dfe2/src/gdsqlite.cpp#L290) to `column_value = Variant((String::utf8((char *)sqlite3_column_text(stmt, i))));` and compiling the plugin fix the issue for me.

**Comment by 2shady4u** [comment]
> I encountered a similar issue. By changing [this line in gdsqlite.cpp](https://github.com/2shady4u/godot-sqlite/blob/70646d844e24015c458c90874986b7c46de1dfe2/src/gdsqlite.cpp#L290) to `column_value = Variant(String::utf8((char *)sqlite3_column_text(stmt, i)));` and compiling the plugin fix the issue for me.

Thanks for finding and reporting this! 😄 I'll look into it as soon as I have some time

**Comment by 2shady4u** [comment]
@lightyears1998 I've seen that you fixed the error on your fork 👀 

Is it working correctly on your end?
If yes, would you be willing to open a pull request with your fix? 😅

**Comment by lightyears1998** [comment]
> @lightyears1998 I've seen that you fixed the error on your fork 👀
> 
> Is it working correctly on your end? If yes, would you be willing to open a pull request with your fix? 😅

It is working correctly on my side. However, I had only tested it for unicode column value. I'd like to do some more additional tests before submitting my PR. It may take a while. 😁

**Comment by 2shady4u** [comment]
This has now been fixed by #90!
